app/Platform/Controllers/RelatedGamesTableController.php

```diff
@@ -54,23 +58,53 @@ public function __invoke(Request $request): View
             });
         }
 
+        if (isset($filterOptions['status'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->gameListService->useGameStatusFilter($game, $filterOptions['status']);
+            });
+        }
+
         $this->gameListService->mergeWantToPlay($loggedInUser);
 
         $this->gameListService->sortGameList($sortOrder);
 
         $availableSorts = $this->gameListService->getAvailableSorts();
-        $availableFilters = [
+        $availableCheckboxFilters = [
             'console' => 'Group by console',
             'populated' => 'Only with achievements',
         ];
+        $availableSelectFilters = [
+            [
+                'kind' => 'status',
+                'label' => 'Status',
+                'options' => [
+                    'all' => 'All games',
+                    'unstarted' => 'No progress',
+                    'unawarded' => 'Unfinished',
+                    'eq-beaten-softcore' => 'Beaten (softcore)',
+                    'eq-beaten-hardcore' => 'Beaten',
+                    'eq-completed' => 'Completed',
+                    'eq-mastered' => 'Mastered',
+                    'any-beaten' => 'Beaten, either hardcore or softcore',
```

Comment:
This should be moved up next to the individual Beaten selections.

Reply:
Renamed internally to `eq-beaten-softcore-or-beaten-hardcore`. The label has been changed to "Beaten, but still missing achievements".

I feel like now with the other changes, this should probably stay put, but will defer to your opinion if you still feel it should be moved:

![Screenshot 2024-01-14 at 11 08 41 AM](https://github.com/RetroAchievements/RAWeb/assets/3984985/b9080b60-1d06-4dbb-a3fc-a0403687d8bb)

Reply:
My brain says all the beaten stuff should be together and all the completed/mastered stuff should be together. "Beaten, but still missing achievements" falls between those two classifications.

I'd probably also move "Has any softcore progress" down to keep the "revised" option near the other completed/mastered options.

Reply:
Adjusted in latest, hopefully this is a bit better:

![Screenshot 2024-01-14 at 3 30 16 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/de443f4e-f80d-47e9-b7d8-2980c1ebfe2b)